src/url/query.zig

```diff
@@ -223,61 +188,244 @@ pub fn parseQuery(alloc: std.mem.Allocator, s: []const u8) !Values {
     return values;
 }
 
-test "parse empty query" {
-    var values = try parseQuery(std.testing.allocator, "");
-    defer values.deinit();
+// The return'd string may or may not be allocated. Callers should use arenas
```

Comment:
Shouldn't we change the allocator we pass to `unescape` in `parseQuery` to use the `values`'s arena then?

Reply:
YES! The test runner doesn't detect memory leaks when using std.testing.allocator right now (https://github.com/lightpanda-io/browser/pull/409)...would have caught that.